gcloud/logging/logger.py

```diff
@@ -204,3 +217,85 @@ def list_entries(self, projects=None, filter_=None, order_by=None,
         return self.client.list_entries(
             projects=projects, filter_=filter_, order_by=order_by,
             page_size=page_size, page_token=page_token)
+
+
+class Batch(object):
+    """Context manager:  collect entries to log via a single API call.
+
+    Helper returned by :meth:Logger.batch
+
+    :type logger: :class:`gcloud.logging.logger.Logger`
+    :param logger: the topic being published
+
+    :type client: :class:`gcloud.logging.client.Client`
+    :param client: The client to use.
+    """
+    def __init__(self, logger, client):
+        self.logger = logger
+        self.entries = []
+        self.client = client
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        if exc_type is None:
+            self.commit()
+
+    def log_text(self, text):
+        """Emulate logging a text entry, but save it.
+
+        :type text: string
+        :param text: the text entry
+        """
+        self.entries.append(('text', text))
+
+    def log_struct(self, info):
+        """Emulate logging a struct entry, but save it.
+
+        :type info: dict
+        :param info: the struct entry
+        """
+        self.entries.append(('struct', info))
+
+    def log_proto(self, message):
+        """Emulate logging a protobuf entry, but save it.
+
+        :type message: protobuf message
+        :param message: the protobuf entry
+        """
+        self.entries.append(('proto', message))
+
+    def commit(self, client=None):
+        """Send saved log entries as a single API call.
+
+        :type client: :class:`gcloud.pubsub.client.Client` or ``NoneType``
+        :param client: the client to use.  If not passed, falls back to the
+                       ``client`` stored on the current batch.
+        """
+        if client is None:
+            client = self.client
+        data = {}
+        entries = data['entries'] = []
+        for entry_type, entry in self.entries:
+            info = {
+                'logName': self.logger.path,
+                'resource': {
+                    'type': 'global',
+                },
+            }
```

Comment:
This info is needed for every single log in the `entries:write` request? :disappointed: